rita/src/rita_client/dashboard/network_endpoints.rs

```diff
@@ -54,24 +58,102 @@ pub fn get_exit_info(_req: HttpRequest) -> Box<Future<Item = Json<Vec<ExitInfo>>
-        .and_then(move |reply| Ok(Json(reply?)))
-        .responder()
+pub fn select_exit(path: Path<String>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    let exit_name = path.into_inner();
+    debug!("/exits/{}/select hit", exit_name);
+
+    let mut exit_client = SETTING.get_exit_client_mut();
+    let mut ret = HashMap::new();
+
+    if exit_client.exits.contains_key(&exit_name) {
+        info!("Selecting exit {:?}", exit_name);
+        exit_client.current_exit = Some(exit_name);
+        return Box::new(future::ok(HttpResponse::Ok().json(ret)));
+    } else {
+        error!("Requested selection of an unknown exit {:?}", exit_name);
+        ret.insert(
+            "error".to_owned(),
+            format!("Requested selection of an unknown exit {:?}", exit_name),
+        );
+        return Box::new(future::ok(
+            HttpResponse::new(StatusCode::BAD_REQUEST)
+                .into_builder()
+                .json(ret),
+        ));
+    }
 }
 
-pub fn select_exit(name: Path<String>) -> Box<Future<Item = Json<()>, Error = Error>> {
-    debug!("/exits/{}/select hit", name);
+pub fn register_to_exit(path: Path<String>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    let exit_name = path.into_inner();
+    debug!("/exits/{}/register hit", exit_name);
+
+    debug!("Attempting to register on exit {:?}", exit_name);
+
+    Box::new(exit_setup_request(exit_name, None).then(|res| {
+        let mut ret = HashMap::new();
+        match res {
+            Ok(_) => future::ok(HttpResponse::Ok().json(ret)),
+            Err(e) => {
+                error!("exit_setup_request() failed with: {:?}", e);
+                ret.insert("error".to_owned(), "Exit setup request failed".to_owned());
+                ret.insert("rust_error".to_owned(), format!("{:?}", e));
+                future::ok(
+                    HttpResponse::new(StatusCode::BAD_REQUEST)
+                        .into_builder()
+                        .json(ret),
+                )
+            }
+        }
+    }))
+}
 
-    Dashboard::from_registry()
-        .send(SelectExit(name.into_inner()))
-        .from_err()
-        .and_then(move |reply| Ok(Json(reply?)))
-        .responder()
+pub fn verify_on_exit_with_code(
+    path: Path<(String, String)>,
+) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    let (exit_name, code) = path.into_inner();
+    debug!("/exits/{}/verify/{} hit", exit_name, code);
+
+    Box::new(exit_setup_request(exit_name, Some(code)).then(|res| {
+        let mut ret = HashMap::new();
+        match res {
+            Ok(_) => future::ok(HttpResponse::Ok().json(ret)),
+            Err(e) => {
+                error!("exit_setup_request() failed with: {:?}", e);
+                ret.insert("error".to_owned(), "Exit setup request failed".to_owned());
+                ret.insert("rust_error".to_owned(), format!("{:?}", e));
+                future::ok(
+                    HttpResponse::new(StatusCode::BAD_REQUEST)
```

Comment:
Same as above - we handle error condition here so perhaps 500+ error would be better?

Reply:
same response